services/cucumber/test/TestCucumberWireProtocolController.cpp

```diff
@@ -0,0 +1,72 @@
+#include "infra/timer/test_helper/ClockFixture.hpp"
+#include "infra/util/BoundedString.hpp"
+#include "services/cucumber/CucumberContext.hpp"
+#include "services/cucumber/CucumberRequestHandlers.hpp"
+#include "services/cucumber/CucumberWireProtocolController.hpp"
+#include "services/cucumber/CucumberWireProtocolParser.hpp"
+#include "services/network/test_doubles/ConnectionMock.hpp"
+#include "gmock/gmock.h"
+#include <algorithm>
+
+namespace
+{
+    class CucumberScenarioRequestHandlerMock
+        : public services::CucumberScenarioRequestHandler
+    {
+    public:
+        MOCK_METHOD(void, BeginScenario, (infra::JsonArray & tags, const infra::Function<void()>& onDone), (override));
+        MOCK_METHOD(void, EndScenario, (const infra::Function<void()>& onDone), (override));
+    };
+
+    class CucumberWireProtocolControllerTest
+        : public testing::Test
+        , public infra::ClockFixture
+    {
+    public:
+        void Request(infra::BoundedConstString request);
+        void RequestWithoutScenarioTags(infra::BoundedConstString request);
+
+    public:
+        services::CucumberContext context;
+        testing::StrictMock<services::ConnectionObserverMock> connectionObserver;
+        testing::StrictMock<CucumberScenarioRequestHandlerMock> scenarionRequestHandler;
+        services::CucumberWireProtocolController controller{ connectionObserver, scenarionRequestHandler };
+        services::CucumberWireProtocolParser parser;
+    };
+
+    void CucumberWireProtocolControllerTest::Request(infra::BoundedConstString request)
+    {
+        parser.ParseRequest(request);
+        controller.HandleRequest(parser);
+    }
+
+    void CucumberWireProtocolControllerTest::RequestWithoutScenarioTags(infra::BoundedConstString request)
+    {
+        parser.ParseRequest(request);
+        parser.scenarioTags = infra::none;
+        controller.HandleRequest(parser);
+    }
+};
+
+MATCHER_P(Equals, value, "")
```

Comment:
Isn't there a testing::Equals available?

Reply:
almost, it is `testing::Eq`, thanks